src/main/java/uk/gov/hmcts/darts/authentication/config/AuthProviderConfigurationProperties.java

```diff
@@ -0,0 +1,32 @@
+package uk.gov.hmcts.darts.authentication.config;
+
+import com.nimbusds.jose.jwk.source.JWKSource;
+import com.nimbusds.jose.jwk.source.JWKSourceBuilder;
+import com.nimbusds.jose.proc.SecurityContext;
+import uk.gov.hmcts.darts.authentication.exception.AuthenticationException;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+
+public interface AuthProviderConfigurationProperties {
+
+    String getAuthorizationUri();
+
+    String getTokenUri();
+
+    String getJwkSetUri();
+
+    String getLogoutUri();
+
+    String getResetPasswordUri();
+
+    default JWKSource<SecurityContext> getJwkSource() {
+        try {
+            URL jwksUrl = new URL(getJwkSetUri());
+
+            return JWKSourceBuilder.create(jwksUrl).build();
```

Comment:
Maybe better to have this in a separate PR. This was changed as the builder is now mandated by nimbus API.  Our previous usage is deprecated. For example:-

new RemoteJWKSet<>(jwksUrl);